File: python/Zetcode/Jython Swing Tutorial/01 - Introduction/quitbutton.py
#!/usr/bin/env jython
# -*- encoding: utf-8 -*-

from java.lang import System
from javax.swing import JFrame
from javax.swing import JPanel
from javax.swing import JButton


class Example(JFrame):
    
    def __init__(self):
        JFrame.__init__(self)
        self.initUI()
        
    def initUI(self):
        panel = JPanel()
        self.getContentPane().add(panel)
        panel.setLayout(None)
        
        qbutton = JButton("Quit", actionPerformed=self.onQuit)
        qbutton.setBounds(50, 60, 80, 30)
        
        panel.add(qbutton)
        
        self.setTitle("Quit Button")
        self.setSize(300, 200)
        self.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE)
        self.setLocationRelativeTo(None)
        self.setVisible(True)
        
    def onQuit(self, e):
        System.exit(0)
        
        
if __name__ == "__main__":
    Example()

File: java/ZetCode/Swing/01 - First Programs/src/QuitButton.java
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

import javax.swing.JButton;
import javax.swing.JFrame;
import javax.swing.JPanel;
import javax.swing.SwingUtilities;


public class QuitButton extends JFrame
{
    public QuitButton()
    {
        initUi();
    }

    public final void initUi()
    {
        JPanel panel = new JPanel();
        this.getContentPane().add(panel);
        panel.setLayout(null);

        JButton quit_button = new JButton("Quit");
        quit_button.setBounds(50, 60, 80, 30);
        quit_button.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                System.exit(0);
            }
        });

        panel.add(quit_button);

        this.setTitle("Quit Button");
        this.setSize(300, 200);
        this.setLocationRelativeTo(null);
        this.setDefaultCloseOperation(EXIT_ON_CLOSE);
    }

    public static void main(String[] args)
    {
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                QuitButton ex = new QuitButton();
                ex.setVisible(true);
            }
        });
    }
}
